qsk-macros/src/analyze.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::{HashMap, HashSet};
+use std::str::FromStr;
+
+use proc_macro_error::{abort, abort_call_site};
+
+use qsk_types::{LayerComposer, LayerRef, KeyCode, ControlCode};
```

Comment:
It's very convenient to use `qsk_types` types for our domain modeling, but it is problematic because it means we can associate proc macro spans with errors that occur in the `lower` and `codegen` stages of proc macro handling (though errors should be more rare in those stages).